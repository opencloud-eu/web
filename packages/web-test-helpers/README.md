# web-test-helpers

This packages provides utilities for unit testing within the OpenCloud app ecosystem.

## Installation

Depending on your package manager, run one of the following commands:

```
$ npm install @opencloud-eu/web-test-helpers --save-dev

$ pnpm add -D @opencloud-eu/web-test-helpers

$ yarn add @opencloud-eu/web-test-helpers --dev
```

Make sure that you have `vue`, `@vue/test-utils`, and `@opencloud-eu/web-pkg` as dev dependencies of your app since those are peer dependencies of this package.
