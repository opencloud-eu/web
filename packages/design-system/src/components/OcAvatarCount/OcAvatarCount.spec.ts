import { shallowMount } from '@opencloud-eu/web-test-helpers'
import Count from './OcAvatarCount.vue'

describe('OcAvatarCount', () => {
  it('dynamically calculates font size', () => {
    const wrapper = shallowMount(Count, {
      props: {
        size: 100,
        count: 2
      }
    })

    expect((wrapper.element as HTMLElement).style.fontSize).toMatch('40px')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
