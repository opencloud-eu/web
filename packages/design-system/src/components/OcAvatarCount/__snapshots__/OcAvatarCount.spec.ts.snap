// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OcAvatarCount > dynamically calculates font size 1`] = `"<span class="oc-avatar-count flex justify-center items-center bg-role-secondary text-role-on-secondary rounded-[50%]" style="width: 100px; height: 100px; font-size: 40px;">+2</span>"`;
